ray-curator/ray_curator/models/clip.py

```diff
@@ -0,0 +1,160 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Model Clips."""
+
+from pathlib import Path
+from typing import Final
+
+import numpy as np
+import numpy.typing as npt
+import torch
+from torchvision import transforms  # type: ignore[import-untyped]
+from transformers import CLIPModel
+
+from .aesthetics import AestheticScorer
+from .base import ModelInterface
+
+_CLIP_MODEL_ID: Final = "openai/clip-vit-large-patch14"
+
+
+class CLIPImageEmbeddings(ModelInterface):
+    """Interface for generating CLIP image embeddings from input images."""
+
+    def __init__(self, model_dir: str) -> None:
+        """Initialize the CLIPImageEmbeddings model."""
+        super().__init__()
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
+        self.dtype = torch.float32
+        self.model_dir = model_dir
+        # These will be initialized in setup()
+        self.clip = None
+        self.transforms = None
+
+    @property
+    def conda_env_name(self) -> str:
+        """Get the conda environment name.
+
+        Returns:
+            The conda environment name.
+
+        """
+        return "video_splitting"
+
+    @property
+    def model_id_names(self) -> list[str]:
+        """Get the model ID names.
+
+        Returns:
+            A list of model IDs used by this model.
+
+        """
+        return [_CLIP_MODEL_ID]
+
+    def setup(self) -> None:
+        """Set up the CLIPImageEmbeddings model."""
+        weight_file = str(Path(self.model_dir) / self.model_id_names[0])
+        self.clip = CLIPModel.from_pretrained(weight_file).to(self.device).eval()
```

Comment:
Same comment as earlier, can we use setup on node to download the weights ? I think when cosmos curate was made, setup on node was not available.

Reply:
will take care this once we refactor every model to use setup_on_node